File: ClassEval_T/cpp/solution/AvgPartition.cpp

#include <vector>

class AvgPartition {
public:
    AvgPartition(const std::vector<int>& lst, int limit);
    std::pair<int, int> set_num() const;
    std::vector<int> get(int index) const;

private:
    std::vector<int> lst;
    int limit;
};
AvgPartition::AvgPartition(const std::vector<int>& lst, int limit)
    : lst(lst), limit(limit) {}

std::pair<int, int> AvgPartition::set_num() const {
    int size = lst.size() / limit;
    int remainder = lst.size() % limit;
    return { size, remainder };
}

std::vector<int> AvgPartition::get(int index) const {
    auto [size, remainder] = set_num();
    int start = index * size + std::min(index, remainder);
    int end = start + size;
    if (index + 1 <= remainder) {
        end += 1;
    }
    return std::vector<int>(lst.begin() + start, lst.begin() + end);
}

File: ClassEval_T/java/solutuon/AvgPartition.java
package org.example;

import java.util.ArrayList;
import java.util.List;

public class AvgPartition {
    private List<Integer> lst;
    private int limit;

    public AvgPartition(List<Integer> lst, int limit) {
        this.lst = lst;
        this.limit = limit;
    }

    public int[] setNum() {
        int size = lst.size() / limit;
        int remainder = lst.size() % limit;
        return new int[]{size, remainder};
    }

    public List<Integer> get(int index) {
        int[] nums = setNum();
        int size = nums[0];
        int remainder = nums[1];
        int start = index * size + Math.min(index, remainder);
        int end = start + size;
        if (index + 1 <= remainder) {
            end += 1;
        }
        return lst.subList(start, end);
    }

    public static void main(String[] args) {
        List<Integer> lst = new ArrayList<>();
        lst.add(1);
        lst.add(2);
        lst.add(3);
        lst.add(4);
        AvgPartition a = new AvgPartition(lst, 2);
        System.out.println(a.get(0));
        System.out.println(a.get(1));
    }
}